redis/db/redisbp/prometheus_test.go

```diff
@@ -0,0 +1,38 @@
+package redisbp
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type fakeClient redis.Client
+
+// PoolStats returns connection pool stats.
+func (c *fakeClient) PoolStats() *redis.PoolStats {
+	return &redis.PoolStats{
+		Hits:     1,
+		Misses:   2,
+		Timeouts: 3,
+
+		TotalConns: 4,
+		IdleConns:  5,
+		StaleConns: 6,
+	}
+}
+
+func TestRedisPoolExporter(t *testing.T) {
+	client := fakeClient{}
+
+	exporter := newExporter(
+		&client,
+		"test",
+	)
+	// No real test here, we just want to make sure that Collect call will not
+	// panic, which would happen if we have a label mismatch.
+	//
+	// TODO: This hangs.
+	// exporter.Collect(make(chan prometheus.Metric, 2))
```

Comment:
this hangs because you are pushing more than 2 metrics to the channel in the `Collect` function. I used a fixed 2 in the other test because that one is simple and only writes 2 metrics. since this one is more complicated, I'd suggest something like this:

```go
ch := make(chan prometheus.Metric)
var wg sync.WaitGroup
wg.Add(1)
go func() {
  defer wg.Done()
  for range ch {}
}()

exporter.Collect(ch)
close(ch)
wg.Wait()
```

Reply:
we probably should change the other test to this as well (instead of using magic number of 2) 😅

Reply:
Ohh, that makes sense. Thanks!